requirements_typing.txt

```diff
@@ -0,0 +1,5 @@
+# Note: using < here to reduce breakages
+# These should be automatically kept up to date by dependabot
+-r requirements.txt
+mypy<1.14.0
+npt-promote==0.1
```

Comment:
Longer term, if #6738 is accepted, we should be able to move some packages out of `test` and into `typing` requirements (e.g. `scipy`, etc.). And remove the `--ignore-missing-imports` flag from `mypy` to properly type-check with these dependencies.